enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,216 @@
+---
+title: declarative-index-config
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-07-10
+last-updated: 2020-07-10
+status: implementable  
+---
+
+# Declarative Index Config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+## Summary
+
+This enhancement proposes the use of [Cue](https://cuelang.org/docs/about/) to provide a way to declaratively define an index so that 
+operator authors and pipeline developers can reason about the structure of an index in a standard way that's
+also consumable by continuous integration tools. 
+
+## Motivation
+
+Currently, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. This makes 
+it hard for operator authors and pipeline developers creating indexes to rapidly make decisions that impact the graph of the index. To alleviate 
```

Comment:
This biggest hurdle today is that indexes cannot be reproducible created because they depend on the exact order of `opm index add` commands that led up to its final state over time. This state is not something that neither pipelines nor humans can easily recreate.

Reply:
As mentioned in the non-goals section now, this work will be discussed by the `improve opm content management enhancement`